CMakeLists.txt

```diff
@@ -223,11 +223,25 @@ if (NOT CMAKE_BUILD_TYPE_UC STREQUAL "RELEASE")
     endif ()
 endif()
 
+if (CMAKE_BUILD_TYPE_UC STREQUAL "RELEASE"
+    OR CMAKE_BUILD_TYPE_UC STREQUAL "RELWITHDEBINFO"
+    OR CMAKE_BUILD_TYPE_UC STREQUAL "MINSIZEREL")
```

Comment:
And while we are at naming ^^, CMake variable "INSTALL_STRIPPED_BINARIES" is another misnomer ... it's not about installing stripped binaries, it controls if the debug symbols shall be splitted from the CH binary. But I can fix that separately.

Reply:
https://github.com/ClickHouse/ClickHouse/pull/38654